docs/contributing/history.md

```diff
@@ -0,0 +1,196 @@
+found in the upstream Argo repository</a>.
+
+
+<table>
+<tr><th>Hera</th><th>Couler</th><th>Argo Python DSL</th></tr>
+<tr>
+
+<td valign="top"><p>
+
+```python
+from hera.workflows import DAG, Container, Parameter, Workflow
+
+with Workflow(
+    generate_name="dag-diamond-",
+    entrypoint="diamond",
+) as w:
+    echo = Container(
+        name="echo",
+        image="alpine:3.7",
+        command=["echo", "{{inputs.parameters.message}}"],
+        inputs=[Parameter(name="message")],
+    )
+    with DAG(name="diamond"):
+        A = echo(name="A", arguments={"message": "A"})
+        B = echo(name="B", arguments={"message": "B"})
+        C = echo(name="C", arguments={"message": "C"})
+        D = echo(name="D", arguments={"message": "D"})
+        A >> [B, C] >> D
+
+w.create()
+```
+
+</p></td>
+
+<td valign="top"><p>
+
+```python
+import couler.argo as couler
+from couler.argo_submitter import ArgoSubmitter
+
+
+def job(name):
+    couler.run_container(
+        image="docker/whalesay:latest",
+        command=["cowsay"],
+        args=[name],
+        step_name=name,
+    )
+
+
+def diamond():
+    couler.dag(
+        [
+            [lambda: job(name="A")],
+            [lambda: job(name="A"), lambda: job(name="B")],  # A -> B
+            [lambda: job(name="A"), lambda: job(name="C")],  # A -> C
+            [lambda: job(name="B"), lambda: job(name="D")],  # B -> D
+            [lambda: job(name="C"), lambda: job(name="D")],  # C -> D
+        ]
+    )
+
+
+diamond()
+submitter = ArgoSubmitter()
+couler.run(submitter=submitter)
+```
+
+</p></td>
+
+<td valign="top"><p>
+
+```python
+from argo.workflows.dsl import Workflow
+
+from argo.workflows.dsl.tasks import *
+from argo.workflows.dsl.templates import *
+
+
+class DagDiamond(Workflow):
+
+    @task
+    @parameter(name="message", value="A")
+    def A(self, message: V1alpha1Parameter) -> V1alpha1Template:
+        return self.echo(message=message)
+
+    @task
+    @parameter(name="message", value="B")
+    @dependencies(["A"])
+    def B(self, message: V1alpha1Parameter) -> V1alpha1Template:
+        return self.echo(message=message)
+
+    @task
+    @parameter(name="message", value="C")
+    @dependencies(["A"])
+    def C(self, message: V1alpha1Parameter) -> V1alpha1Template:
+        return self.echo(message=message)
+
+    @task
+    @parameter(name="message", value="D")
+    @dependencies(["B", "C"])
+    def D(self, message: V1alpha1Parameter) -> V1alpha1Template:
+        return self.echo(message=message)
+
+    @template
+    @inputs.parameter(name="message")
+    def echo(self, message: V1alpha1Parameter) -> V1Container:
+        container = V1Container(
+            image="alpine:3.7",
+            name="echo",
+            command=["echo", "{{inputs.parameters.message}}"],
+        )
+
+        return container
+```
+
+</p></td>
+</tr>
+</table>
+</details>
+
+## Hera V5 vs V4
+
+_Reserving here for Bloomberg history with Argo/Hera._
```

Comment:
🚀

Reply:
Will write something up internally and run it by @samj1912, will add in another PR